morpheus-graphql-client/src/Data/Morpheus/Client.hs

```diff
@@ -58,11 +60,17 @@ defineByDocumentFile filePath args = do
   qAddDependentFile filePath
   defineByDocument (L.readFile filePath) args
 
+defineByDocumentFile' :: Q FilePath -> (ExecutableDocument, String) -> Q [Dec]
```

Comment:
hi @AlistairB. thanks :) that cool. i really like idea!

but i don't like naming. since mostly `'` in haskell means strickt evaluation. 
moreover it does not explains what it is for. can we give them more descriptive names?

 ````hs
  defineByDocumentFileWithRelativePath ::
  defineByIntrospectionFileWithRelativePath :: 
  ````
 
or we can just use type-classes 

```hs
class defineBy path where 
   defineByIntrospectionFile :: path -> (ExecutableDocument, String) -> Q [Dec]
   defineByDocumentFile :: path -> (ExecutableDocument, String) -> Q [Dec]
```

what do you think?

Reply:
I don't like a solution like `defineByIntrospectionFileWithRelativePath` as:

* it's not flexible, since it just gives you 2 ways to generate the file path. Where exposing `Q FilePath` means the user can generate it in any way they want.
* it would add a dependency on `file-embed`, when many users would not need this functionality.

I think a typeclass is overkill for the problem as well.

I would either expose 2 version have I have done in this PR, which you can decide the naming of.

The apostrophe thing is interesting, in some cases the apostrophe variant is strict, indeed the kowainik guide refers to it as such https://kowainik.github.io/posts/naming-conventions#apostrophe . However, I think it actually more generally refers to a [variant of the function](https://www.reddit.com/r/haskell/comments/kcsi7l/kowainik_foo_to_bar_naming_conventions_in_haskell/gfsta4o/?utm_source=reddit&utm_medium=web2x&context=3).

Alternatively, I think a breaking change updating the `defineByDocumentFile` functions to take a `Q FilePath` would also be fine, since doing `pure "mydocumentfile.gql"` would not be a huge deal.

Anyway, happy to go with whatever solution you like.

Reply:
@AlistairB , then I think  `defineByDocumentFile'` okey. but please then add Haddock (documentation comments) to clarify what it is for and how to use it. 
 